executor/src/witgen/global_constraints.rs

```diff
@@ -279,9 +281,12 @@ fn propagate_constraints<T: FieldElement>(
             }
 
             // Detect [ x ] in [ RANGE ], where RANGE is in the full span.
-            // In that case, we can remove the lookup, because its only function is to enforce
-            // the range constraint.
-            if receive.selected_tuple.expressions.len() == 1 {
+            // In that case, we can remove the bus interaction pair,
```

Comment:
Do we actually remove the full pair? We only remove the send, right? There could be other sends on the same bus.

Reply:
That is a good point, actually. They are never removed and that is also correct, because the same receive might also handle a conditional range constraint (e.g. if you have `[b1] in [BYTES]; cond $ [b2] in [BYTES]`). Which means I can also remove the handling of the multiplicity columns of removed receives :)

Reply:
But shouldn't you change the comment to say that we can remove the send?

Reply:
I mean this is the meaning of the return value of this function